clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1545,6 +1605,30 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
     break;
   }
   case Builtin::BI__builtin_hlsl_elementwise_saturate:
+  case Builtin::BI__builtin_hlsl_select: {
```

Comment:
This addition splits the existing common case code for `BI__builtin_hlsl_elemetwise_saturate` and `BI__builtin_hlsl_elemetwise_rcp` - and would result in incorrect behavior for `BI__builtin_hlsl_elemetwise_saturate`.